packages/reactive-element/src/decorators/query.ts

```diff
@@ -59,50 +60,63 @@ export type QueryDecorator = {
 export function query(selector: string, cache?: boolean): QueryDecorator {
   return (<C extends Interface<ReactiveElement>, V extends Element>(
     protoOrTarget: ClassAccessorDecoratorTarget<C, V>,
-    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>
+    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>,
+    descriptor?: PropertyDescriptor
   ) => {
     const doQuery = (el: Interface<ReactiveElement>): V => {
+      // TODO: if we want to allow users to assert that the query will never
+      // return null, we need a new option and to throw here if the result
+      // is null.
       return (el.renderRoot?.querySelector(selector) ?? null) as V;
     };
-    if (typeof nameOrContext === 'object') {
+    if (cache) {
+      // Accessors to wrap from either:
+      //   1. The decorator target, in the case of standard decorators
+      //   2. The property descriptor, in the case of experimental decorators
+      //      on auto-accessors.
+      //   3. Functions that access our own cache-key property on the instance,
+      //      in the case of experimental decorators on fields.
+      const {get, set} =
```

Comment:
Worth renaming `get` and `set` so it's clear that this is getting/setting the cache?

Reply:
renaming requires more syntax, that might be harder to read:

```ts
const {get: getValue, set: Value} = 
```